### PR TITLE
#984: Update form error messages on domain management contacts

### LINKED ISSUE #984: Domain Mgmt: Refine error messages to be more specific
### Issue Description

Individual error messages appear at the top of the page when a field is left blank. But—on some pages of the Domain Mgmt section—those messages are generic ("This field is required"). Therefore, the individual messages are not useful. The following pages have generic error messages: 

- Authorizing Official
- Your Contact Information

Image

### Additional Context (optional)

Update the error messages to be more specific to the required field on these pages: 

- DNS name servers
- Authorizing Official
- Your Contact Information

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #984 

## Changes

- Overrides error messages for Contact Form on domain management page. Error messages updated to reflect the [content doc](https://docs.google.com/document/d/1EHMzgmig6vaSFRvCiih87Cx38lD2rWrkEE4BzDzev9Q/edit) (also linked in the issue).
- Creates a child class of ContactForm called AuthorizingOfficialContactForm which specifies custom error messages for Authorizing Official page of domain management page. 

**Your contact information error messages**
image

**Authorizing official error messages**
image

## Setup

Go to [staging link](https://getgov-es.app.cloud.gov) and go to manage domain.

- Navigate to Your contact information page and enter empty inputs for required fields. The form should output error messages for the specific fields that were not entered.
- Navigate to Authorizing official page and repeat the same steps as above.

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [x] Created/modified automated tests
- [x] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [x] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by therealslimhsiehdy — CHANGES_REQUESTED** [review]
Maybe I am confused, but I think for Authorizing Official there maybe different errors? This seems like the correct errors for "Your contact information" 😅 

If I'm incorrect, feel free to dismiss the request changes!

File: src/registrar/forms/domain.py [hunk 1]
```diff
@@ -148,11 +148,41 @@ def __init__(self, *args, **kwargs):
...
+
+
+class AuthorizingOfficialContactForm(ContactForm):
+    """Form for updating authorizing official contacts."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Set custom error messages
+        self.fields["first_name"].error_messages = {"required": "Enter the first name / given name of this contact."}
```
[the remark this answers is not in the document]
Comment (erinysong) [reply src/registrar/forms/domain.py:170]: hm interesting, you're right that the domain management form for authorizing official has different error messages than the application form. we should make sure that they're consistent and we can ask content which one is preferred. @michelle-rago do you have a preference between which of the 2 error message formats we use for authorizing officials?:

1. Enter the first name / given name of this contact. (what was requested on this [content doc](https://docs.google.com/document/d/1EHMzgmig6vaSFRvCiih87Cx38lD2rWrkEE4BzDzev9Q/edit#heading=h.gjdgxs) and issue)
2. Enter the first name / given name of your authorizing official. (what is currently used on the domain application)

File: src/registrar/forms/domain.py [hunk 2]
```diff
@@ -148,11 +148,41 @@ def __init__(self, *args, **kwargs):
...
+            "required": "Enter an email address in the required format, like name@example.com."
+        }
+        self.fields["phone"].error_messages = {"required": "Enter a phone number for this contact."}
+
 
 class DomainSecurityEmailForm(forms.Form):
     """Form for adding or editing a security email to a domain."""
 
-    security_email = forms.EmailField(label="Security email", required=False)
+    security_email = forms.EmailField(label="Security email (optional)", required=False)
```
[the remark this answers is not in the document]
Comment (erinysong) [reply src/registrar/forms/domain.py]: Leftover code from testing something else - thank you for catching that! just removed it

File: src/registrar/forms/domain.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (erinysong) [reply src/registrar/forms/domain.py]: Resolved - everything should be updated to authorizing official text (option 2)

**Review by therealslimhsiehdy — APPROVED** [review]
Looks great for contact + authorizing official! 

Screenshot 2023-11-21 at 9 59 57 AM

Screenshot 2023-11-21 at 9 59 44 AM

File: src/registrar/forms/domain.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (michelle-rago) [reply src/registrar/forms/domain.py]: I know I'm late on this...just wanted to say "thanks" for picking the more specific set of error messages!

**Pull request merged**